scripts/machine-launch-script.sh

```diff
@@ -1,85 +1,271 @@
+    else
+	wget -O install_conda.sh "$CONDA_INSTALLER"  || curl -fsSLo install_conda.sh "$CONDA_INSTALLER"
+	echo "::INFO:: installing conda to '$CONDA_INSTALL_PREFIX'"
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+
+        # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+        # for more information on strict channel_priority
+        $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+        # By default, don't mess with people's PS1, I personally find it annoying
+        $SUDO "$CONDA_EXE" config --system --set changeps1 false
+
+	conda_init_extra_args=()
+	if [[ -n "$SUDO"  || "$USER" == root ]]; then
+	    # if we're installing into a root-owned directory using sudo, or we're already root
+	    # initialize conda in the system-wide rcfiles
+	    conda_init_extra_args=(--no-user --system)
+	fi
+	$SUDO "${CONDA_EXE}" init $DRY_RUN_OPTION "${conda_init_extra_args[@]}" bash
+    fi
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync )
+    # cross-compile glibc 2.28+ deps
+    CONDA_PACKAGE_SPECS+=( make )
+    # other misc deps
+    # conda-forge splits out the shared library into libzlib
+    # not sure if we need both, but I'm including them for now
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+	libzlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect
```

Comment:
Stupid shell scripting and missing newline escapes.
```suggestion
        expect \
```